### ISSUE TITLE
[FEATURE REQUEST] put monitor workspace after last primary workspace

### ISSUE THREAD

**Issue by kkga** [fthx/simple-task-bar#8]
Note: this issue happens when GNOME multi-monitor handling is set to "Workspaces on primary display only".

Icons for apps that are placed on the secondary display appear on the left of the workspaces list. Not sure if that's intended design, but it feels and looks weird.

![image](https://user-images.githubusercontent.com/1460122/86495045-5ce20000-bd80-11ea-88bc-5d6d04ebb10c.png)

I think, ideally, i'd rather not see the icons for apps on secondary display at all since they're not attached to any workspace.
Alternatively, maybe show them at the very end of the task bar instead, after some sort of visual divider.

**Comment by fthx** [comment]
I work very often with a 2nd display.

The icons of the 2nd display's apps have to be displayed if I want the extension to to its job. These "sticky windows" cannot easily be displayed on a per-workspace taskbar.

So, why not placing them at the beginning ? This behaviour is intended. Though, I can make a setting to place them at the end. But, as you mentioned it, I have to find a logical separator. I can imagine sort of monitor icon in the same frame as the primary display workspaces. Tell me. From my POV, it's not great : the last GNOME workspace is void son it would make two separators side-by-side.

What do you think of a monitor icon before sticky windows, at the left (as it's right now), but with a monitor icon ?

**Comment by fthx** [comment]
Please note that you do use fixed workspaces number but GNOME's default is dynamic.

**Comment by kkga** [comment]
Yeah I do use static workspaces. Got so used to it that I forgot it’s not the default 😅

You’re right that not displaying the icons would be an odd choice. However it’s also odd that “secondary” apps are displayed at the very beginning of the task bar imho.

Not sure what exactly you mean regarding the last workspace being “void”. Can you clarify?

If you’ll go for showing them at the end of the list, I think simply a vertical line divider would be enough. Putting a display icon might be a bit too heavy.

**Comment by fthx** [comment]
(Running default gnome settings, the last dynamically created workspace is void.)
Ok. I'm definitely not convinced with putting sticky zone at the end. Does not look good to me. So I stay with the current order. I created two new settings: 1) display or not the monitor label 2) set this label.
You can find it here, before it's reviewed:
https://extensions.gnome.org/review/download/17787.shell-extension.zip
(review page: https://extensions.gnome.org/review/17787 )

Though, you can easily place the monitor windows at the end, by cut-pasting the piece of code starting with:
`// create sticky workspace icon + all sticky windows (on all workspaces) icons and buttons`
after the part starting with:
`// create all normal windows icons and buttons`

I let this issue open. If some users request the same feature, I'll make a placement setting for monitor zone. Please note that I can't name "Monitor" this zone since there are sticky windows (on all workspaces) that are not displayed in your external monitor (e.g. Google Meet pop-up for presentation indicator).
